soccer/Processor.cpp

```diff
@@ -590,6 +602,74 @@ void Processor::run() {
     vision.stop();
 }
 
+/*
+ * Updates the geometry packet if different from the existing one,
+ * Based on the geometry vision data.
+ */
+void Processor::updateGeometryPacket(const SSL_GeometryFieldSize* fieldSize) {
+    if (fieldSize->field_lines_size() == 0) {
+        return;
+    }
+
+    const SSL_FieldCicularArc* penalty = nullptr;
+    const SSL_FieldCicularArc* center = nullptr;
+    float displacement =
+        Field_Dimensions::Default_Dimensions.GoalFlat();  // default displacment
+
+    for (int i = 0; i < fieldSize->field_arcs().size(); i++) {
+        if (fieldSize->field_arcs().Get(i).name() == "CenterCircle") {
+            // Assume center circle
+            center = &fieldSize->field_arcs().Get(i);
+        } else if (fieldSize->field_arcs().Get(i).name() ==
+                   "LeftFieldLeftPenaltyArc") {
+            penalty = &fieldSize->field_arcs().Get(i);
+        }
+    }
+
+    for (int i = 0; i < fieldSize->field_lines().size(); i++) {
+        if (fieldSize->field_lines().Get(i).name() == "RightPenaltyStretch") {
+            displacement = abs(fieldSize->field_lines().Get(i).p2().y() -
+                               fieldSize->field_lines().Get(i).p1().y());
+        }
+    }
+
+    float thickness = fieldSize->field_lines().Get(0).thickness() / 1000.0f;
+
+    // The values we get are the center of the lines, we want to use the
+    // outside, so we can add this as an offset.
+    float adj = fieldSize->field_lines().Get(0).thickness() / 1000.0f / 2.0f;
+
+    float fieldBorder = currentDimensions->Border();
+
+    if (penalty != nullptr && center != nullptr && thickness != 0) {
+        // Force a resize
+        Field_Dimensions newDim = Field_Dimensions(
+            fieldSize->field_length() / 1000.0f,
+            fieldSize->field_width() / 1000.0f, fieldBorder, thickness,
+            fieldSize->goal_width() / 1000.0f,
+            fieldSize->goal_depth() / 1000.0f,
+            Field_Dimensions::Default_Dimensions.GoalHeight(),
+            penalty->radius() / 1000.0f + adj,  // PenaltyDist
+            Field_Dimensions::Default_Dimensions.PenaltyDiam(),
+            penalty->radius() / 1000.0f + adj,       // ArcRadius
+            center->radius() / 1000.0f + adj,        // CenterRadius
+            (center->radius()) * 2 / 1000.0f + adj,  // CenterDiameter
+            displacement / 1000.0f,                  // GoalFlat
+            (fieldSize->field_length() / 1000.0f + (fieldBorder)*2),
+            (fieldSize->field_width() / 1000.0f + (fieldBorder)*2));
+
+        if (newDim != *currentDimensions) {
+            // Set the changed field dimensions to the current ones
+            cerr << "Updating field geometry based off of vision packet."
```

Comment:
We don't have much of a logging system setup, although it would be nice!  See #435